### PR TITLE
[language] Include ContractEvent in writeset transaction

### PR DESCRIPTION
**Pull request by runtian-zhou**
## Motivation

This PR will enable ContractEvents to be included in the writeset transaction. By doing so we would be able to include reconfiguration events in the genesis transaction.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

## Related PRs

#1383

### CODE REVIEW

File: types/src/transaction/write_set.rs [hunk 1]
```diff
@@ -0,0 +1,25 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{contract_event::ContractEvent, write_set::WriteSet};
+use serde::{Deserialize, Serialize};
+
+#[derive(Clone, Debug, Hash, Eq, PartialEq, Serialize, Deserialize)]
+pub struct WriteSetPayload {
```
Comment (runtian-zhou) [inline types/src/transaction/write_set.rs]: Any suggestions for a better name here? It seems to me that calling it a `ChangeSet` makes more sense but I would like to hear your thoughts.

Comment (davidiw) [reply types/src/transaction/write_set.rs]: I think that's fine

File: types/src/transaction/write_set.rs [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    }
+
+    pub fn into_inner(self) -> (WriteSet, Vec<ContractEvent>) {
+        (self.write_set, self.events)
+    }
+
+    pub fn write_set(&self) -> &WriteSet {
+        &self.write_set
+    }
+}
```
Comment (sblackshear) [inline types/src/transaction/write_set.rs]: Getter for `events` too?

File: types/src/transaction/write_set.rs (hunk 1 above)
Comment (sblackshear) [reply types/src/transaction/write_set.rs]: `TransactionEffect` is another idea.

**Pull request closed without merging**